### PR TITLE
Allow node_groups to be a nested dict

### PR DESCRIPTION
**Pull request by thesquelched**
Allow `node_groups` to be either a `list` of `dicts`, or a single, nested `dict` with the node group ID's as keys.

### CODE REVIEW

File: lavaclient/api/clusters.py [hunk 1]
```diff
@@ -251,9 +263,11 @@ def create(self, name, stack_id, username=None, ssh_keys=None,
...
                          the key `user@hostname`, creating the key from
                          `$HOME/.ssh/id_rsa.pub` if it doesn't exist.
-        :param node_groups: List of node group dictionaries for the cluster
-                            for which you want to modify the stack defaults,
-                            e.g. the node count or the flavor
+        :param node_groups: `dict` of `(node_group_id, attrs)` pairs, in which
+                            `attrs` is a `dict` of node group attributes.
+                            Instead of a `dict`, you may give a `list` of
+                            `dicts`, each containing the `id` key. Currently
+                            supported attributes are `flavor_id` and `count`
```
Comment (pchanne) [inline lavaclient/api/clusters.py]: an example here would make it more clear.

Comment (thesquelched) [inline lavaclient/api/clusters.py]: An example wouldn't be appropriate in a docstring.  I'll put an example in the docs in a future PR.

Comment (pchanne) [inline lavaclient/api/clusters.py]: Okay. That works too.

**Comment by pchanne** [comment]
+1

**Comment by pchanne** [comment]
Oh wait, isn't there any cli test that needs to be updated?

**Comment by thesquelched** [comment]
No, this doesn't apply to the CLI, as it will only ever pass in the original 'list of dicts' format.

**Pull request merged**